dockerfiles/QAT/run.sh

```diff
@@ -0,0 +1,149 @@
+testsrepopath=${GOPATH}/src/${testsrepo}
+runtimerepo=github.com/kata-containers/runtime
+runtimerepopath=${GOPATH}/src/${runtimerepo}
+agentrepo=github.com/kata-containers/agent
+agentrepopath=${GOPATH}/src/${agentrepo}
+
+grab_kata_repos()
+{
+	# Check out all the repos we will use now, so we can try and ensure they use the specified branch
+	# Only check out the branch needed, and make it shallow and thus space/bandwidth efficient
+	git clone --single-branch --branch $KATA_VERSION --depth=1 https://${packagerepo} ${packagerepopath}
+	git clone --single-branch --branch $KATA_VERSION --depth=1 https://${osbrepo} ${osbrepopath}
+	git clone --single-branch --branch $KATA_VERSION --depth=1 https://${testsrepo} ${testsrepopath}
+	git clone --single-branch --branch $KATA_VERSION --depth=1 https://${runtimerepo} ${runtimerepopath}
+	git clone --single-branch --branch $KATA_VERSION --depth=1 https://${agentrepo} ${agentrepopath}
+}
+
+configure_kernel()
+{
+	cp /input/qat.conf ${packagerepopath}/kernel/configs/fragments/common/qat.conf
+	# We need yq and go to grab kernel versions etc.
+	${testsrepopath}/.ci/install_yq.sh
+	${testsrepopath}/.ci/install_go.sh -p
+	cd ${packagerepopath}
+	./kernel/build-kernel.sh setup
+}
+
+build_kernel()
+{
+	cd ${packagerepopath}
+	LINUX_VER=$(ls -d kata-linux-*)
+	sed -i 's/EXTRAVERSION =/EXTRAVERSION = .qat.container/' $LINUX_VER/Makefile
+	./kernel/build-kernel.sh build
+}
+
+build_rootfs()
+{
+	cd ${osbrepopath}/rootfs-builder
+	# We default to using clearlinux for the QAT rootfs
+	SECCOMP=no EXTRA_PKGS='kmod' ./rootfs.sh clearlinux
+}
+
+grab_qat_drivers()
+{
+	mkdir -p $QAT_SRC
+	cd $QAT_SRC
+	curl -L $QAT_DRIVER_URL | tar zx
+}
+
+build_qat_drivers()
+{
+	cd ${packagerepopath}
+	linux_kernel_path=${packagerepopath}/${LINUX_VER}
+	KERNEL_MAJOR_VERSION=$(awk '/^VERSION =/{print $NF}' ${linux_kernel_path}/Makefile)
+	KERNEL_PATHLEVEL=$(awk '/^PATCHLEVEL =/{print $NF}' ${linux_kernel_path}/Makefile)
+	KERNEL_SUBLEVEL=$(awk '/^SUBLEVEL =/{print $NF}' ${linux_kernel_path}/Makefile)
+	KERNEL_EXTRAVERSION=$(awk '/^EXTRAVERSION =/{print $NF}' ${linux_kernel_path}/Makefile)
+	KERNEL_ROOTFS_DIR=${KERNEL_MAJOR_VERSION}.${KERNEL_PATHLEVEL}.${KERNEL_SUBLEVEL}${KERNEL_EXTRAVERSION}
+	cd $QAT_SRC
+	KERNEL_SOURCE_ROOT=${linux_kernel_path} ./configure ${QAT_COMPILE_OPTIONS}
+	make all -j$(nproc)
+}
+
+add_qat_to_rootfs()
+{
+	cd $QAT_SRC
+	ROOTFS_DIR=${osbrepopath}/rootfs-builder/rootfs-Clear
+	make INSTALL_MOD_PATH=$ROOTFS_DIR qat-driver-install -j$(nproc)
+
+	cp $QAT_SRC/build/usdm_drv.ko $ROOTFS_DIR/usr/lib/modules/${KERNEL_ROOTFS_DIR}/updates/drivers
+	depmod -a -b ${ROOTFS_DIR} ${KERNEL_ROOTFS_DIR}
+	cd ${osbrepopath}/image-builder
+	./image_builder.sh ${ROOTFS_DIR}
+}
+
+copy_outputs()
+{
+	mkdir -p ${OUTPUT_DIR}
+	cp ${linux_kernel_path}/arch/x86/boot/bzImage $OUTPUT_DIR/vmlinuz-${LINUX_VER}_qat
+	cp ${osbrepopath}/image-builder/kata-containers.img $OUTPUT_DIR
+}
+
+help() {
+	usage=$(cat << EOF
```

Comment:
Nit you could just do:


```sh
cat <<EOT

  blah, blah

EOT
```

... to avoid creating the variable.

Reply:
done